api_v3/services/ServerNodeService.php

```diff
@@ -0,0 +1,147 @@
+		$serverNode->fromObject($dbServerNode, $this->getResponseProfile());
+		return $serverNode;
+	}
+	
+	/**
+	 * Get server node by id
+	 * 
+	 * @action get
+	 * @param int $serverNodeId
+	 * @throws KalturaErrors::INVALID_OBJECT_ID
+	 * @return KalturaServerNode
+	 */
+	function getAction($serverNodeId)
+	{
+		$dbServerNode = ServerNodePeer::retrieveByPK($serverNodeId);
+		if (!$dbServerNode)
+			throw new KalturaAPIException(KalturaErrors::INVALID_OBJECT_ID, $serverNodeId);
+		
+		$serverNode = KalturaServerNodeFactory::getInstanceByType($dbServerNode->getType());
+		$serverNode->fromObject($dbServerNode, $this->getResponseProfile());
+		return $serverNode;
+	}
+	
+	/**
+	 * Update server node by id 
+	 * 
+	 * @action update
+	 * @param int $serverNodeId
+	 * @param KalturaServerNode $serverNode
+	 * @return KalturaServerNode
+	 */
+	function updateAction($serverNodeId, KalturaServerNode $serverNode)
+	{
+		$dbServerNode = ServerNodePeer::retrieveByPK($serverNodeId);
+		if (!$dbServerNode)
+			throw new KalturaAPIException(KalturaErrors::INVALID_OBJECT_ID, $serverNodeId);
+			
+		$dbServerNode = $serverNode->toUpdatableObject($dbServerNode);
+		$dbServerNode->save();
+		
+		$serverNode = KalturaServerNodeFactory::getInstanceByType($dbServerNode->getType());
+		$serverNode->fromObject($dbServerNode, $this->getResponseProfile());
+		return $serverNode;
+	}
+	
+	/**
+	 * delete server node by id
+	 *
+	 * @action delete
+	 * @param string $serverNodeId
+	 * @throws KalturaErrors::INVALID_OBJECT_ID
+	 */
+	function deleteAction($serverNodeId)
+	{
+		$dbServerNode = ServerNodePeer::retrieveByPK($serverNodeId);
+		if(!$dbServerNode)
+			throw new KalturaAPIException(KalturaErrors::INVALID_OBJECT_ID, $serverNodeId);
+	
+		$dbServerNode->setStatus(ServerNodeStatus::DELETED);
+		$dbServerNode->save();
+	}
+	
+	/**	
+	 * @action list
+	 * @param KalturaServerNodeFilter $filter
+	 * @param KalturaFilterPager $pager
+	 * @return KalturaServerNodeListResponse
+	 */
+	public function listAction(KalturaServerNodeFilter $filter = null, KalturaFilterPager $pager = null)
+	{
+		if(!$filter)
+			$filter = new KalturaServerNodeFilter();
+			
+		if(!$pager)
+			$pager = new KalturaFilterPager();
+		
+		return $filter->getTypeListResponse($pager, $this->getResponseProfile(), null);
+	}
+	
+	/**
+	 * Update server node status
+	 *
+	 * @action reportStatus
+	 * @param string $hostName
+	 * @return KalturaServerNode
+	 */
+	function reportStatusAction($hostName, KalturaServerNode $serverNode = null)
+	{
+		$dbServerNode = ServerNodePeer::retrieveByPartnerIdAndHostName($this->getPartnerId(), $hostName);
+		
+		//Allow serverNodes auto registration without calling add
+		if (!$dbServerNode)
+		{
+			if($serverNode)
+			{
+				return $this->addAction($serverNode);
+			}
+			else 
+				throw new KalturaAPIException(KalturaErrors::SERVER_NODE_NOT_FOUND, $hostName);
+		}
+	
+		$dbServerNode->setHeartbeatTime(time());
```

Comment:
I think you should also call toUpdateable in order to apply status attributes from $serverNode to $dbServerNode.